ldap/servers/slapd/daemon.c

```diff
@@ -222,7 +222,27 @@ disk_mon_get_mount_point(char *dir)
         }
         if (s.st_dev == dev_id) {
             endmntent(fp);
-            return (slapi_ch_strdup(mnt->mnt_dir));
+            if ((strncmp(mnt->mnt_dir, "/dev", 4) == 0 ||
```

Comment:
I think the logic here looks wrong? Shouldn't it be:
```
if (
    (strncmp(mnt->mnt_dir, "/dev", 4) == 0 && strncmp(mnt->mnt_dir, "/dev/shm", 8) != 0) ||
    strncmp(mnt->mnt_dir, "/proc", 4) == 0 ||
    strncmp(mnt->mnt_dir, "/sys", 4) == 0)

```

Reply:
You get the same result, but yours is clearer.

Reply:
Cool, lets go for clarity then, otherwise, ack :)

Reply:
Agreed, changes made, and merging...